components/aws/sagemaker/tests/unit_tests/tests/commonv2/test_sagemaker_component_v2.py

```diff
@@ -0,0 +1,437 @@
+        mock_values.extra_param = "How did this even get here?"  # type: ignore
+
+        self.component._write_all_outputs(mock_paths, mock_values)
+
+        mock_logging.error.assert_called_once()
+        self.component._write_output.assert_has_calls(
+            [
+                call("/tmp/output1", "value1", json_encode=False),
+                call("/tmp/output2", ["value2"], json_encode=True),
+            ]
+        )
+
+    def test_write_output(self):
+        mock_output_path = "/tmp/output1"
+
+        with patch("commonv2.sagemaker_component.Path", MagicMock()) as mock_path:
+            self.component._write_output(mock_output_path, "value1")
+
+            mock_path.assert_has_calls(
+                [
+                    call(mock_output_path),
+                    call().parent.mkdir(parents=True, exist_ok=True),
+                    call(mock_output_path),
+                    call().write_text("value1"),
+                ]
+            )
+
+        with patch("commonv2.sagemaker_component.Path", MagicMock()) as mock_path:
+            self.component._write_output(mock_output_path, "value2", json_encode=True)
+
+            mock_path().write_text.assert_any_call('"value2"')
+
+        with patch("commonv2.sagemaker_component.Path", MagicMock()) as mock_path:
+            self.component._write_output(mock_output_path, ["value1"], json_encode=True)
+
+            mock_path().write_text.assert_any_call(json.dumps(["value1"]))
+
+    @patch.object(SageMakerComponent, "_get_k8s_api_client", MagicMock())
+    @patch("kubernetes.client.CoreV1Api")
+    def test_init_configure_k8s(self, mock_k8s_api):
+
+        self.component.namespace = "test"
+
+        self.component._init_configure_k8s()
+
+        mock_k8s_api.assert_called_once()
+        mock_k8s_api().list_namespaced_pod.assert_called_once_with(namespace="test")
+
+    def test_get_current_namespace(self):
+
+        with patch("os.path.exists", MagicMock(return_value=True)):
+            with patch(
+                "builtins.open",
+                MagicMock(
+                    return_value=open("tests/unit_tests/tests/commonv2/files/namespace")
+                ),
+            ) as mock_open:
+                self.assertEqual(
+                    self.component._get_current_namespace(), "namespace-in-file"
+                )
+
+        with patch("os.path.exists", MagicMock(return_value=False)):
+
+            with patch(
+                "kubernetes.config.list_kube_config_contexts"
+            ) as mock_k8s_context:
+                mock_k8s_context.return_value = (
+                    " ",
+                    {"context": {"namespace": "default-ns"}},
+                )
+                self.component._get_current_namespace()
+                self.assertEqual("default-ns", self.component._get_current_namespace())
+
+    @patch.object(SageMakerComponent, "_get_k8s_api_client", MagicMock())
+    def test_wait_resource_consumed_by_controller(self):
```

Comment:
nit: This is a general comment for several of the tests in this file, looks like several of the methods that are tested here have varying return values i.e(True/False, name-spaced or not). In some of these tests its a little hard to read what is going on and quickly discern what the test is about. Maybe its worth splitting some of these longer tests into seperate subtests for readibility sake

Reply:
Will split and add comments.